File: Networking/country_ip_lookup/ip_lookup.go
package main

import (
	"fmt"
  "log"
  "net/http"
  "encoding/json"
  "io/ioutil"
)

type JSON struct {
  Country string `json:"country"`
}

func main() {
  fmt.Println("Please enter the IP address...")
  var ip string
  fmt.Scan(&ip)
	response, err := http.Get("http://ip-api.com/json/" + ip)
  if err != nil {
     log.Fatalln(err)
  }
  data, err := ioutil.ReadAll(response.Body)
  if err != nil {
    log.Fatal(err)
  }
  var country JSON
  json.Unmarshal(data, &country)
  fmt.Println(country.Country)
}

File: Networking/country_ip_lookup/ip_lookup_js/ip_lookup.js
// looks up the country of an IP address 
// this app requires a package please install it useing NPM I 

const https = require('https');
const readlineSync = require("readline-sync"); 

let ip = readlineSync.question("Enter the IP address you want to look up: ");

https.get(`https://ipapi.co/${ip}/json/`, (res) => {
    res.setEncoding('utf8');
    let body = '';
    res.on('data', (data) => {
        body += data;
    });
    res.on('end', () => {
        let parsed = JSON.parse(body);
        console.log(`The country of ${ip} is ${parsed.country_name}`);
    });
}).on('error', (err) => {
    console.log(`Error: ${err.message}`);
});